### PR TITLE
drivers: video: Update some features

### PR DESCRIPTION
**Pull request by SPRESENSE**
## Summary
Update video driver with isx012 and isx019 image sensor
- Add missing FAR qualifier
- Add get frame interval API
- Update API to dump register for debug
- isx012: Support clip feature
- isx019: Increase waiting time for HW initialization
- isx019: Fix white balance control and capability
- isx019: Fix incorrect JPEG DQT table

## Impact
video driver.

## Testing
spresense:example_camera

### CODE REVIEW

**Review by jerpelea — COMMENTED** [review]
@SPRESENSE please split the drivers/video/isx019: Fix and update isx019 driver patch in 3 separate patches

**Review by jerpelea — APPROVED** [review]

**Comment by acassis** [comment]
Please fix the CI reported error:
```
drivers/video/isx019.c:1109:38: error: Operator/assignment must be followed with whitespace
```

**Comment by SPRESENSE** [comment]
I addd a commit changed not to use `timercmp()` that caused the CI error. please review again.

File: drivers/video/isx012.c [hunk 1]
```diff
@@ -2150,14 +2300,57 @@ static int isx012_get_value(uint32_t id,
...
+}
+
 static int isx012_set_value(uint32_t id,
                             uint32_t size,
-                            FAR imgsensor_value_t value)
+                            imgsensor_value_t value)
 {
   FAR struct isx012_dev_s *priv = &g_isx012_private;
   int       ret = -EINVAL;
   uint8_t   cnt;
```
Comment (pkarashchenko) [inline drivers/video/isx012.c:2352]:
```suggestion
  int           ret = -EINVAL;
  uint8_t       cnt;
```

File: drivers/video/isx012.c [hunk 2]
```diff
@@ -1827,7 +1977,7 @@ static int isx012_get_value(uint32_t id,
   uint8_t    cnt;
   uint8_t    threea_enable;
   uint16_t   read_src;
```
Comment (pkarashchenko) [inline drivers/video/isx012.c:1979]:
```suggestion
  uint8_t        cnt;
  uint8_t        threea_enable;
  uint16_t       read_src;
```

File: drivers/video/isx012.c [hunk 3]
```diff
@@ -1827,7 +1977,7 @@ static int isx012_get_value(uint32_t id,
...
-  uint16_t   *read_dst;
+  FAR uint16_t *read_dst;
   int        ret = OK;
   uint16_t   exposure_time_lsb;
   uint16_t   exposure_time_msb;
```
Comment (pkarashchenko) [inline drivers/video/isx012.c:1983]:
```suggestion
  int            ret = OK;
  uint16_t       exposure_time_lsb;
  uint16_t       exposure_time_msb;
```

File: drivers/video/isx012.c [hunk 4]
```diff
@@ -841,6 +988,11 @@ static int isx012_set_mode_param(isx012_dev_t *priv,
   uint16_t vsize_addr;
   uint8_t  smode;
   uint8_t  mode;
```
Comment (pkarashchenko) [inline drivers/video/isx012.c:990]:
```suggestion
  uint8_t smode;
  uint8_t mode;
```

File: drivers/video/isx012.c [hunk 5]
```diff
@@ -825,7 +845,134 @@ static bool is_movie_needed(uint8_t fmt, uint8_t fps)
...
+static void activate_clip(FAR isx012_dev_t *priv,
+                          uint16_t w,
+                          uint16_t h,
+                          FAR isx012_rect_t *clip)
+{
+  uint8_t  hvfree = 0;
+  uint32_t r_x = ZOOM_UNIT;
+  uint32_t r_y = ZOOM_UNIT;
+  int32_t  x   = 0;
+  int32_t  y   = 0;
```
Comment (pkarashchenko) [inline drivers/video/isx012.c:934]: Minor, optional: either
```
  uint8_t  hvfree = 0;
  uint32_t r_x     = ZOOM_UNIT;
  uint32_t r_y     = ZOOM_UNIT;
  int32_t  x       = 0;
  int32_t  y       = 0;
```
or
```
  uint8_t  hvfree = 0;
  uint32_t r_x = ZOOM_UNIT;
  uint32_t r_y = ZOOM_UNIT;
  int32_t  x = 0;
  int32_t  y = 0;
```

File: drivers/video/isx019.c [hunk 6]
```diff
@@ -1442,6 +1461,12 @@ static int activate_clip(imgsensor_stream_type_t type,
           }
 
         break;
+
+      default: /* Otherwise, clear clip setting. */
+        size = FPGA_CLIP_NON;
+        top  = 0;
+        left = 0;
+        break;
```
Comment (pkarashchenko) [inline drivers/video/isx019.c:1469]:
```suggestion
        left = 0;

        break;
```

**Review by pkarashchenko — COMMENTED** [review]
Only one important comment in `try_repeat`. All other are minor and optional

File: drivers/video/isx019.c [hunk 7]
```diff
@@ -1503,30 +1528,27 @@ static int isx019_start_capture(imgsensor_stream_type_t type,
...
+
+      default: /* 160 */
+        regval |= FPGA_SCALE_160_120;
+        break;
+    }
+
+  activate_clip(type,
+               fmt[IMGSENSOR_FMT_MAIN].width,
+               fmt[IMGSENSOR_FMT_MAIN].height);
```
[the remark this answers is not in the document]
Comment (SPRESENSE) [reply drivers/video/isx019.c]: Fixed.

File: drivers/video/isx019.c [hunk 8]
```diff
@@ -2128,7 +2226,53 @@ static int set_wbmode(imgsensor_value_t val)
...
+  return OK;
+}
+
+static bool is_awb_enable(void)
+{
+  uint8_t mode = AWBMODE_AUTO;
+
+  isx019_i2c_read(CAT_CATAWB, AWBMODE, &mode, 1);
+
+  return (mode == AWBMODE_HOLD) ? false : true;
```
[the remark this answers is not in the document]
Comment (SPRESENSE) [reply drivers/video/isx019.c]: Fixed.

File: drivers/video/video.c [hunk 9]
```diff
@@ -402,10 +402,10 @@ static video_parameter_name_t g_video_parameter_name[] =
...
 
 static FAR void *video_handler;
-static FAR const struct imgsensor_ops_s **g_video_registered_sensor;
-static int g_video_registered_sensor_num;
-static FAR const struct imgsensor_ops_s *g_video_sensor_ops;
-static FAR const struct imgdata_ops_s *g_video_data_ops;
+static FAR const struct imgsensor_ops_s **g_video_registered_sensor = NULL;
+static int g_video_registered_sensor_num = 0;
+static FAR const struct imgsensor_ops_s *g_video_sensor_ops = NULL;
+static FAR const struct imgdata_ops_s *g_video_data_ops = NULL;
```
[the remark this answers is not in the document]
Comment (SPRESENSE) [reply drivers/video/video.c]: Fixed.

File: drivers/video/isx012.c [hunk 10]
```diff
@@ -2150,14 +2300,57 @@ static int isx012_get_value(uint32_t id,
...
+                            imgsensor_value_t value)
 {
   FAR struct isx012_dev_s *priv = &g_isx012_private;
   int       ret = -EINVAL;
   uint8_t   cnt;
-  uint16_t  *write_src;
+  FAR uint16_t *write_src;
   uint16_t  write_dst;
   uint16_t  regval;
   uint16_t  exposure_time_lsb;
```
[the remark this answers is not in the document]
Comment (SPRESENSE) [reply drivers/video/isx012.c]: Fixed.

File: drivers/video/isx012.c (hunk 1 above)
Comment (SPRESENSE) [reply drivers/video/isx012.c]: Fixed.

File: drivers/video/isx012.c (hunk 2 above)
Comment (SPRESENSE) [reply drivers/video/isx012.c]: Fixed.

File: drivers/video/isx012.c (hunk 3 above)
Comment (SPRESENSE) [reply drivers/video/isx012.c]: Fixed.

File: drivers/video/isx012.c (hunk 5 above)
Comment (SPRESENSE) [reply drivers/video/isx012.c]: Fixed.

File: drivers/video/isx019.c (hunk 6 above)
Comment (SPRESENSE) [reply drivers/video/isx019.c]: Fixed.

**Review by pkarashchenko — APPROVED** [review]

**Comment by jerpelea** [comment]
@SPRESENSE can you please do the changes

**Pull request merged**